x/market/keeper/key.go

```diff
@@ -43,6 +44,25 @@ func leaseKey(id types.LeaseID) []byte {
 	return buf.Bytes()
 }
 
+func leaseKeyActive(id types.LeaseID) []byte {
+	buf := bytes.NewBuffer(leaseActivePrefix)
+	buf.Write(id.Owner.Bytes())
+	binary.Write(buf, binary.BigEndian, id.DSeq)
+	binary.Write(buf, binary.BigEndian, id.GSeq)
+	binary.Write(buf, binary.BigEndian, id.OSeq)
+	buf.Write(id.Provider.Bytes())
+	return buf.Bytes()
+}
+
+func convertLeaseActiveKey(activeKey []byte) ([]byte, error) {
+	buf := bytes.NewBuffer(leasePrefix)
+	_, err := buf.Write(activeKey[2:])
```

Comment:
i think here and also in the active group index it might be best to use the length of the key.

```go
	_, err := buf.Write(activeKey[len(leaseActivePrefix):])
```

Reply:
Oh I like that better!